packages/yarnpkg-core/rollup.config.js

```diff
@@ -0,0 +1,45 @@
+import cjs                  from "@rollup/plugin-commonjs";
+import resolve              from "@rollup/plugin-node-resolve";
+import path                 from "path";
+import esbuild              from "rollup-plugin-esbuild";
+import {brotliCompressSync} from "zlib";
+
+function wrapOutput() {
+  return {
+    name: `wrap-output`,
+    generateBundle(options, bundle, isWrite) {
+      const bundles = Object.keys(bundle);
+      if (bundles.length !== 1)
+        throw new Error(`Expected only one bundle, got ${bundles.length}`);
+      const outputBundle = bundle[bundles[0]];
+
+      outputBundle.code = `let hook;\n\nmodule.exports = () => {\n  if (typeof hook === \`undefined\`)\n    hook = require('zlib').brotliDecompressSync(Buffer.from('${brotliCompressSync(
+        outputBundle.code.replace(/\r\n/g, `\n`)
+      ).toString(`base64`)}', 'base64')).toString();\n\n  return hook;\n};\n`;
+    },
+  };
+}
+
+// eslint-disable-next-line arca/no-default-export
+export default [
+  {
+    input: `./sources/ZipConvertWorker.ts`,
+    output: {
+      file: `./sources/ZipConvertWorkerSource.js`,
+      format: `cjs`,
+      strict: false,
+      preferConst: true,
+    },
+    plugins: [
+      resolve({
+        extensions: [`.mjs`, `.js`, `.ts`, `.tsx`, `.json`],
+        rootDir: path.join(__dirname, `../../`),
+        jail: path.join(__dirname, `../../`),
+        preferBuiltins: true,
+      }),
+      esbuild({tsconfig: false, target: `es2018`}),
+      cjs({transformMixedEsModules: true, extensions: [`.js`, `.ts`]}),
+      wrapOutput(),
+    ],
+  },
+];
```

Comment:
Could we just use esbuild?

Reply:
Should be doable but rollup is better at tree shaking as far as I can tell

Reply:
I'd prefer to use the same bundler as much as possible; we already use webpack and eslint, and webpack is on its way out. Better to unify this logic.